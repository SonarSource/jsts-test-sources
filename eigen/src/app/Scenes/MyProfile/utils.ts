/**
 *
 * @param {string} bio
 * @returns {string} A properly formatted bio for the collector profile
 * @example
 * normalizeMyProfileBio(
 * `Sentence 1\nSentence 2\rSentence 3`) = "Sentence 1. Sentence 2. Sentence 3"
 */
export const normalizeMyProfileBio = (bio: string) =>
  bio
    .replace(/(\r\n|\n|\r)/gm, ". ") // Replace line breaks with ". "
    .replace("..", ".") // Avoid double dots between sentences after replacing line breaks with ". "
