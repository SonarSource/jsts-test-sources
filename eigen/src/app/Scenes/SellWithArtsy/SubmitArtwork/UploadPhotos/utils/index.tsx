import { LocationResult } from "app/Components/DetailedLocationAutocomplete"
import { goBack, switchTab } from "app/navigation/navigate"
import { BottomTabType } from "app/Scenes/BottomTabs/BottomTabType"
import { SellWithArtsyHomeQueryRenderer } from "app/Scenes/SellWithArtsy/SellWithArtsyHome"
import { GlobalStore } from "app/store/GlobalStore"
import React, { useCallback, useEffect, useRef } from "react"
import { BackHandler } from "react-native"
import { Image as RNCImage } from "react-native-image-crop-picker"

/** The metadata for a consigned work */
export interface SellWithArtsyMetadata {
  title: string | null
  year: string | null
  category: string | null
  categoryName: string | null
  medium: string | null
  width: string | null
  height: string | null
  depth: number | null
  unit: string | null
  displayString: string | null // This would look something like "1/5", "5/5"
}

export interface ArtistResult {
  internalID: string
  name: string
  image?: {
    url: string
  }
  targetSupply?: {
    isTargetSupply: boolean
  }
}

export type SearchResult = LocationResult | ArtistResult

export interface Photo {
  image: RNCImage
  uploaded?: boolean
  uploading?: boolean
}

export interface SellWithArtsySetup {
  submissionID?: string
  state?: "DRAFT" | "SUBMITTED"
  artist?: ArtistResult
  photos?: Photo[]
  metadata?: SellWithArtsyMetadata
  provenance?: string
  editionInfo?: {
    size?: string
    number?: string
  }
  utmMedium?: string
  utmSource?: string
  utmTerm?: string
  signed?: boolean
  certificateOfAuth?: boolean
  location?: {
    city: string
    state: string
    country: string
  }
  editionScreenViewed?: boolean
}

export interface SellTabProps {
  overwriteHardwareBackButtonPath?: BottomTabType
}

export const SellWithArtsy: React.FC = () => {
  const sellTabProps = GlobalStore.useAppState((state) => {
    return state.bottomTabs.sessionState.tabProps.sell ?? {}
  }) as SellTabProps

  const overwriteHardwareBackButtonPath = sellTabProps?.overwriteHardwareBackButtonPath ?? null

  const sellTabPropsRef = useRef<BottomTabType | null>(null)

  useEffect(() => {
    sellTabPropsRef.current = overwriteHardwareBackButtonPath
  }, [overwriteHardwareBackButtonPath])

  useEffect(
    useCallback(() => {
      BackHandler.addEventListener("hardwareBackPress", handleBackButton)

      return () => BackHandler.removeEventListener("hardwareBackPress", handleBackButton)
    }, [])
  )

  const handleBackButton = useCallback(() => {
    if (sellTabPropsRef.current) {
      switchTab(sellTabPropsRef.current)
      GlobalStore.actions.bottomTabs.setTabProps({ tab: "sell", props: {} })
    } else {
      goBack()
    }

    return true
  }, [sellTabPropsRef.current])

  return <SellWithArtsyHomeQueryRenderer />
}
