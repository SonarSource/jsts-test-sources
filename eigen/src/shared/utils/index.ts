export * from "./ArtsyKeyboardAvoidingView"
export * from "./MeasuredView"
export * from "./Wrap"
export * from "./flattenChildren"
export * from "./normalizeText"
