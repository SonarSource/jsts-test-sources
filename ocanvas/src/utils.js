(function(oCanvas, window, document, undefined){

	// Extend an object with new properties and replace values for existing properties
	oCanvas.extend = function () {
	
		// Get first two args
		var args = Array.prototype.slice.call(arguments),
			last = args[args.length - 1],
			destination = args.splice(0, 1)[0],
			current = args.splice(0, 1)[0],
			x, exclude = [],
			descriptor;
		
		// If the last object is an exclude object, get the properties
		if (last.exclude && (JSON.stringify(last) === JSON.stringify({exclude:last.exclude}))) {
			exclude = last.exclude;
		}
		
		// Do the loop unless this object is an exclude object
		if (current !== last || exclude.length === 0) {
			
			// Add members from second object to the first
			for (x in current) {
			
				// Exclude specified properties
				if (~exclude.indexOf(x)) {
					continue;
				}
				
				descriptor = Object.getOwnPropertyDescriptor(current, x);
				
				if (descriptor.get || descriptor.set) {
					Object.defineProperty(destination, x, descriptor);
				} else {
					destination[x] = current[x];
				}
			}
		}
		
		// If there are more objects passed in, run once more, otherwise return the first object
		if (args.length > 0) {
			return oCanvas.extend.apply(this, [destination].concat(args));
		} else {
			return destination;
		}
	};

	oCanvas.addDOMEventHandler = function (core, domObject, eventName, handler, useCapture) {
		core.domEventHandlers.push({
			obj: domObject,
			event: eventName,
			handler: handler,
			useCapture: !!useCapture
		});
		domObject.addEventListener(eventName, handler, useCapture);
	};

	oCanvas.removeDOMEventHandler = function (core, index) {
		var data = core.domEventHandlers[index];
		data.obj.removeEventListener(data.event, data.handler, data.useCapture);
	};

	oCanvas.isNumber = function (n) {
		return !isNaN(parseFloat(n)) && isFinite(n);
	};

	// Define Object.create if not available
	if (typeof Object.create !== "function") {
		Object.create = function (o) {
			function F() {}
			F.prototype = o;
			return new F();
		};
	}

	// Define Object.getPropertyDescriptor if not available.
	// This function will check for the descriptor in the whole prototype chain.
	if (typeof Object.getPropertyDescriptor !== "function") {
		Object.getPropertyDescriptor = function(object, property) {
			var descriptor = Object.getOwnPropertyDescriptor(object, property);
			var proto = Object.getPrototypeOf(object);
			while (descriptor === undefined && proto !== null) {
				descriptor = Object.getOwnPropertyDescriptor(proto, property);
				proto = Object.getPrototypeOf(proto);
			}
			return descriptor;
		};
	}

	// requestAnimationFrame polyfill by Erik MÃ¶ller
	// fixes from Paul Irish and Tino Zijdel
	// http://paulirish.com/2011/requestanimationframe-for-smart-animating/
	// http://my.opera.com/emoller/blog/2011/12/20/requestanimationframe-for-smart-er-animating
	(function () {
		var lastTime = 0;
		var vendors = ['ms', 'moz', 'webkit', 'o'];
		for (var x = 0; x < vendors.length && !window.requestAnimationFrame; ++x) {
			window.requestAnimationFrame = window[vendors[x] + 'RequestAnimationFrame'];
			window.cancelAnimationFrame = window[vendors[x] + 'CancelAnimationFrame']
			  || window[vendors[x] + 'CancelRequestAnimationFrame'];
		}

		if (!window.requestAnimationFrame) {
			window.requestAnimationFrame = function (callback, element) {
				var currTime = new Date().getTime();
				var timeToCall = Math.max(0, 16 - (currTime - lastTime));
				var id = window.setTimeout(function () { callback(currTime + timeToCall); }, timeToCall);
				lastTime = currTime + timeToCall;
				return id;
			};
		}

		if (!window.cancelAnimationFrame) {
			window.cancelAnimationFrame = function (id) {
				clearTimeout(id);
			};
		}
	}());

	// usage: log('inside coolFunc',this,arguments);
	// http://paulirish.com/2009/log-a-lightweight-wrapper-for-consolelog/
	window.log = function () {
		log.history = log.history || [];	// store logs to an array for reference
		log.history.push(arguments);
		if (this.console) {
			var i, args = Array.prototype.slice.call(arguments), l = args.length;
			for (i = 0; i < l; i++) {
				console.log(args[i]);
			}
		}
	};

})(oCanvas, window, document);
