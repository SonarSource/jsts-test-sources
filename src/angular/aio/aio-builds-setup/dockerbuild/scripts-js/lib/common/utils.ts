// Functions
export const assertNotMissingOrEmpty = (name: string, value: string | null | undefined) => {
  if (!value) {
    throw new Error(`Missing or empty required parameter '${name}'!`);
  }
};

export const getEnvVar = (name: string, isOptional = false): string => {
  const value = process.env[name];

  if (!isOptional && !value) {
    console.error(`ERROR: Missing required environment variable '${name}'!`);
    process.exit(1);
  }

  return value || '';
};
