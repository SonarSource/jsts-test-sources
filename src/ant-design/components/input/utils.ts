import type { ClearableInputProps } from './ClearableLabeledInput';
import type { InputProps } from './Input';

// eslint-disable-next-line import/prefer-default-export
export function hasPrefixSuffix(props: InputProps | ClearableInputProps) {
  return !!(props.prefix || props.suffix || props.allowClear);
}
