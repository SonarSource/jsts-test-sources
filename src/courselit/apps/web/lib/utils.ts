// const fs = require("fs");
// const { spawn } = require("child_process");
// const { v4: uuidv4 } = require("uuid");
//
export const capitalize = (s: string) => {
    if (typeof s !== "string") return "";
    return s.charAt(0).toUpperCase() + s.slice(1);
};
//
// exports.foldersExist = (folders) => {
//   for (const folder of folders) {
//     if (!fs.existsSync(folder)) {
//       return false;
//     }
//   }
//
//   return true;
// };
//
// exports.createFolders = (folders) => {
//   for (const folder of folders) {
//     if (!fs.existsSync(folder)) {
//       fs.mkdirSync(folder, { recursive: true });
//     }
//   }
// };
//
// /**
//  * A pure function to generate a string by appending current epoch
//  * to the provided filename.
//  *
//  * @param {string} filename
//  */
// exports.uniqueFileNameGenerator = (filename) => {
//   const extention = filename.split(".");
//   // const uniqueNameWithoutExtention = `${extention.slice(
//   //   0,
//   //   extention.length - 1
//   // )}_${Date.now()}`;
//
//   return {
//     name: uuidv4(),
//     ext: extention[extention.length - 1],
//   };
// };
//
// /**
//  * A wrapper to promisify the move function of express-upload.
//  *
//  * @param {object} file - The express-upload file object
//  * @param {string} path - Where to move the current file
//  */
// exports.moveFile = (file, path) =>
//   new Promise((resolve, reject) => {
//     file.mv(path, (err) => {
//       if (err) reject(err.message);
//
//       resolve();
//     });
//   });
//
// /**
//  * Promisifies command line utility cwebp.
//  *
//  * @param {string} path - file path of the file to be converted
//  * @param {quality} quality - a number representing quality of the output. 0 is worst and 100 is best.
//  */
// exports.convertToWebp = (path, quality = 75) =>
//   new Promise((resolve, reject) => {
//     const process = spawn(
//       "cwebp",
//       [`"${path}"`, `-o "${path}"`, `-q ${quality}`],
//       {
//         shell: true,
//       }
//     );
//
//     process.on("exit", (code) => {
//       if (code !== 0) {
//         reject(new Error("Error in converting the file to Webp format."));
//       }
//
//       resolve();
//     });
//   });
//
// /**
//  * Async handler for async middlewares.
//  *
//  * Copied from: https://stackoverflow.com/questions/61086833/async-await-in-express-middleware
//  */
// exports.asyncHandler = (fn) => (req, res, next) =>
//   Promise.resolve(fn(req, res, next)).catch(next);
//
// exports.getParentDirectory = (path) => {
//   const filePathSplit = path.split("/");
//   filePathSplit.pop();
//   return filePathSplit.join("/");
// };
//
export const isSubscriptionValid = (dateStr: Date): boolean => {
    return new Date(dateStr).getTime() > new Date().getTime();
};
//
interface MagicLinkProps {
    token: string;
    hostname: string;
    secure: boolean;
    redirect?: string;
}
export const generateMagicLink = ({
    token,
    hostname,
    secure = true,
    redirect,
}: MagicLinkProps) => {
    return `http${secure ? "s" : ""}://${hostname}/login?token=${token}${
        redirect ? `&redirect=${redirect}` : ""
    }`;
};
