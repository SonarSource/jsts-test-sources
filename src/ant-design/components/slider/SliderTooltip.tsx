import raf from 'rc-util/lib/raf';
import { composeRef } from 'rc-util/lib/ref';
import * as React from 'react';
import { useRef } from 'react';
import type { TooltipProps } from '../tooltip';
import Tooltip from '../tooltip';

const SliderTooltip = React.forwardRef<unknown, TooltipProps>((props, ref) => {
  const { visible } = props;
  const innerRef = useRef<any>(null);

  const rafRef = useRef<number | null>(null);

  function cancelKeepAlign() {
    raf.cancel(rafRef.current!);
    rafRef.current = null;
  }

  function keepAlign() {
    rafRef.current = raf(() => {
      innerRef.current?.forcePopupAlign();
      rafRef.current = null;
    });
  }

  React.useEffect(() => {
    if (visible) {
      keepAlign();
    } else {
      cancelKeepAlign();
    }

    return cancelKeepAlign;
  }, [visible, props.title]);

  return <Tooltip ref={composeRef(innerRef, ref)} {...props} />;
});

export default SliderTooltip;
