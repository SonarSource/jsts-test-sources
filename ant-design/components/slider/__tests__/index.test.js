import { mount } from 'enzyme';
import React from 'react';
import Slider from '..';
import focusTest from '../../../tests/shared/focusTest';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';
import { sleep } from '../../../tests/utils';
import ConfigProvider from '../../config-provider';
import SliderTooltip from '../SliderTooltip';

describe('Slider', () => {
  mountTest(Slider);
  rtlTest(Slider);
  focusTest(Slider, { testLib: true });

  it('should show tooltip when hovering slider handler', () => {
    const wrapper = mount(<Slider defaultValue={30} />);
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseEnter');
    expect(wrapper.find('Trigger').render()).toMatchSnapshot();
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseLeave');
    expect(wrapper.find('Trigger').render()).toMatchSnapshot();
  });

  it('should show correct placement tooltip when set tooltipPlacement', () => {
    const wrapper = mount(<Slider vertical defaultValue={30} tooltipPlacement="left" />);
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseEnter');
    expect(wrapper.find('Trigger').render()).toMatchSnapshot();
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseLeave');
    expect(wrapper.find('Trigger').render()).toMatchSnapshot();
  });

  it('when tooltipVisible is true, tooltip should show always, or should never show', () => {
    let wrapper = mount(<Slider defaultValue={30} tooltipVisible />);
    expect(wrapper.find('.ant-tooltip-content').at(0).hasClass('ant-tooltip-hidden')).toBe(false);
    wrapper.find('.ant-slider-handle').at(0).simulate('mouseEnter');
    expect(wrapper.find('.ant-tooltip-content').at(0).hasClass('ant-tooltip-hidden')).toBe(false);
    wrapper.find('.ant-slider-handle').at(0).simulate('click');
    expect(wrapper.find('.ant-tooltip-content').at(0).hasClass('ant-tooltip-hidden')).toBe(false);
    wrapper = mount(<Slider defaultValue={30} tooltipVisible={false} />);
    expect(wrapper.find('.ant-tooltip-content').length).toBe(0);
  });

  it('when step is null, thumb can only be slided to the specific mark', () => {
    const intentionallyWrongValue = 40;
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(
      <Slider marks={marks} defaultValue={intentionallyWrongValue} step={null} tooltipVisible />,
    );
    expect(wrapper.find('.ant-slider-handle').get(0).props).toHaveProperty('aria-valuenow', 48);
  });

  it('when step is not null, thumb can be slided to the multiples of step', () => {
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(<Slider marks={marks} defaultValue={49} step={1} tooltipVisible />);
    expect(wrapper.find('.ant-slider-handle').get(0).props).toHaveProperty('aria-valuenow', 49);
  });

  it('when step is undefined, thumb can be slided to the multiples of step', () => {
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(
      <Slider marks={marks} defaultValue={49} step={undefined} tooltipVisible />,
    );
    expect(wrapper.find('.ant-slider-handle').get(0).props).toHaveProperty('aria-valuenow', 49);
  });

  it('should render in RTL direction', () => {
    const wrapper = mount(
      <ConfigProvider direction="rtl">
        <Slider defaultValue={30} tooltipVisible />
      </ConfigProvider>,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should keepAlign by calling forcePopupAlign', async () => {
    let ref;
    mount(
      <SliderTooltip
        title="30"
        visible
        ref={node => {
          ref = node;
        }}
      />,
    );
    ref.forcePopupAlign = jest.fn();
    await sleep(20);
    expect(ref.forcePopupAlign).toHaveBeenCalled();
  });

  it('tipFormatter should not crash with undefined value', () => {
    [undefined, null].forEach(value => {
      mount(<Slider value={value} tooltipVisible />);
    });
  });
  it('step should not crash with undefined value', () => {
    [undefined, null].forEach(value => {
      mount(<Slider step={value} tooltipVisible />);
    });
  });
});
