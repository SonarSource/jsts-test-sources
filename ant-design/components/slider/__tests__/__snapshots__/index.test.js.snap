// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-slider ant-slider-rtl ant-slider-horizontal"
>
  <div
    class="ant-slider-rail"
  />
  <div
    class="ant-slider-track"
    style="right: 0%; width: 0%;"
  />
  <div
    class="ant-slider-step"
  />
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="0"
    class="ant-slider-handle"
    role="slider"
    style="right: 0%; transform: translateX(50%);"
    tabindex="0"
  />
</div>
`;

exports[`Slider should render in RTL direction 1`] = `
<div
  class="ant-slider ant-slider-rtl ant-slider-horizontal"
>
  <div
    class="ant-slider-rail"
  />
  <div
    class="ant-slider-track"
    style="right: 0%; width: 30%;"
  />
  <div
    class="ant-slider-step"
  />
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="ant-slider-handle ant-tooltip-open"
    role="slider"
    style="right: 30%; transform: translateX(50%);"
    tabindex="0"
  />
  <div>
    <div
      class="ant-tooltip ant-slider-tooltip ant-tooltip-rtl ant-zoom-down-appear ant-zoom-down-appear-prepare ant-zoom-down"
      style="opacity: 0;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          30
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider should show correct placement tooltip when set tooltipPlacement 1`] = `
Array [
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="ant-slider-handle ant-tooltip-open"
    role="slider"
    style="bottom: 30%; transform: translateY(50%);"
    tabindex="0"
  />,
  <div>
    <div
      class="ant-tooltip ant-slider-tooltip ant-zoom-down-appear ant-zoom-down-appear-prepare ant-zoom-down"
      style="opacity: 0;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          30
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Slider should show correct placement tooltip when set tooltipPlacement 2`] = `
Array [
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="ant-slider-handle"
    role="slider"
    style="bottom: 30%; transform: translateY(50%);"
    tabindex="0"
  />,
  <div>
    <div
      class="ant-tooltip ant-slider-tooltip ant-zoom-down-leave ant-zoom-down-leave-start ant-zoom-down"
      style="pointer-events: none;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          30
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Slider should show tooltip when hovering slider handler 1`] = `
Array [
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="ant-slider-handle ant-tooltip-open"
    role="slider"
    style="left: 30%; transform: translateX(-50%);"
    tabindex="0"
  />,
  <div>
    <div
      class="ant-tooltip ant-slider-tooltip ant-zoom-down-appear ant-zoom-down-appear-prepare ant-zoom-down"
      style="opacity: 0;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          30
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Slider should show tooltip when hovering slider handler 2`] = `
Array [
  <div
    aria-disabled="false"
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="30"
    class="ant-slider-handle"
    role="slider"
    style="left: 30%; transform: translateX(-50%);"
    tabindex="0"
  />,
  <div>
    <div
      class="ant-tooltip ant-slider-tooltip ant-zoom-down-leave ant-zoom-down-leave-start ant-zoom-down"
      style="pointer-events: none;"
    >
      <div
        class="ant-tooltip-content"
      >
        <div
          class="ant-tooltip-arrow"
        >
          <span
            class="ant-tooltip-arrow-content"
          />
        </div>
        <div
          class="ant-tooltip-inner"
          role="tooltip"
        >
          30
        </div>
      </div>
    </div>
  </div>,
]
`;
